zipline/finance/performance/position.py

```diff
@@ -167,14 +173,18 @@ def adjust_commission_cost_basis(self, sid, cost):
             return
 
         prev_cost = self.cost_basis * self.amount
-        new_cost = prev_cost + cost
+        if isinstance(asset, Future):
```

Comment:
I am closer to :-1: than :1: on making the change to Equity to include a multiplier value.

- There are places where we do dispatch because `Equities`and `Futures` behave differently. 
e.g. cases like `period.PerformancePeriod.handle_execution` where that value should be a no-op for `Equity`s. I believe in most cases we do want to dispatch on Asset type because we want to consider whether the contract contributes to exposure, should calculate a payout, etc.
- `multiplier` is something that is inherit to the definition of a future contract; adding it to an `Equity` object would be clouding the modelling of an equity which may lead to regressions in treating Assets alike in cases where they should not be.

Needless to say, 'not for this PR', but have we considered modelling the portfolio by having an `EquityPortfolio` and a `FuturePortfolio`, where the overall `Portfolio` is compromised of the two?
This may allow us to model the unique behaviors of each more cleanly.